app/uk/gov/hmrc/vatregisteredcompanies/repositories/PayloadBufferRepository.scala

```diff
@@ -16,61 +16,75 @@
 
 package uk.gov.hmrc.vatregisteredcompanies.repositories
 
+import com.mongodb.client.model.InsertOneOptions
+import org.bson.types.ObjectId
+
 import javax.inject.{Inject, Singleton}
-import play.api.Logger
+import play.api.Logging
 import play.api.libs.json._
-import play.modules.reactivemongo.ReactiveMongoComponent
-import reactivemongo.api.Cursor
-import reactivemongo.bson.{BSONDocument, BSONObjectID}
-import reactivemongo.play.json.ImplicitBSONHandlers._
-import uk.gov.hmrc.mongo.ReactiveRepository
-import uk.gov.hmrc.vatregisteredcompanies.models.Payload
+import org.mongodb.scala.bson.{BsonDocument}
+import uk.gov.hmrc.mongo.play.json.formats.MongoFormats.Implicits.objectIdFormat
+import org.mongodb.scala.model.Filters.equal
+import org.mongodb.scala.model.Indexes.ascending
+import org.mongodb.scala.model.{Filters, FindOneAndDeleteOptions, IndexModel, IndexOptions, InsertOneOptions}
+import uk.gov.hmrc.mongo.MongoComponent
+import uk.gov.hmrc.mongo.play.json.PlayMongoRepository
+import uk.gov.hmrc.vatregisteredcompanies.models.{Payload}
 
 import scala.concurrent.{ExecutionContext, Future}
 
-final case class PayloadWrapper(
-  _id: BSONObjectID = BSONObjectID.generate,
+final case class PayloadWrapper (
+  _id: ObjectId = ObjectId.get(),
   payload: Payload
 )
 
 object PayloadWrapper {
-  val format: Format[PayloadWrapper] = Json.format[PayloadWrapper]
+  implicit  val format: Format[PayloadWrapper] = Json.format[PayloadWrapper]
+
 }
 
 @Singleton
 class   PayloadBufferRepository@Inject()(
-  reactiveMongoComponent: ReactiveMongoComponent
+  mongoComponent: MongoComponent
 )(implicit val executionContext: ExecutionContext)
-  extends ReactiveRepository(
-    "vatregisteredcompaniesbuffer",
-    reactiveMongoComponent.mongoConnector.db,
-    PayloadWrapper.format) {
+  extends PlayMongoRepository[PayloadWrapper](
+    mongoComponent = mongoComponent,
+    collectionName = "vatregisteredcompaniesbuffer",
+    domainFormat = PayloadWrapper.format,
+    indexes = Seq()) with Logging {
 
   def deleteAll(): Future[Unit] =
-    removeAll().map(_=> ())
+    collection.deleteMany(Filters.empty()).toFuture().map(_ => ())
 
-  implicit val format: OFormat[PayloadWrapper] = Json.format[PayloadWrapper]
+  //implicit val format: OFormat[PayloadWrapper] = Json.format[PayloadWrapper]
 
   def insert(payload: Payload): Future[Unit] =
-    collection.insert(true).one(PayloadWrapper(payload = payload)).map(_ => ())
+    collection.insertOne(PayloadWrapper(payload = payload)).toFuture().map(_ => ())
 
   def list: Future[List[PayloadWrapper]] =
-    findAll()
+    collection.find(Filters.empty()).toFuture().map(_.toList)
 
   private def getOne: Future[List[PayloadWrapper]] = {
-    val query = BSONDocument()
+    val query = BsonDocument()
```

Comment:
Can we do Filters.empty() here?

Reply:
yes you can use Filters.empty() instead of BSONDocument() if you prefer that

Reply:
Completed.